### PR TITLE
Remove v7x checks for sparse vector mappings

### PR DESCRIPTION
**Pull request by benwtrent**
removing old 7x checks for v9 upgrade.

### CODE REVIEW

**Review by john-wagster — APPROVED** [review]
LGTM

**Review by carlosdelest — APPROVED** [review]
👍 🧹

File: server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java [hunk 1]
```diff
@@ -58,8 +57,6 @@ public class SparseVectorFieldMapper extends FieldMapper {
 
     public static final String CONTENT_TYPE = "sparse_vector";
 
-    static final String ERROR_MESSAGE_7X = "[sparse_vector] field type in old 7.x indices is allowed to "
-        + "contain [sparse_vector] fields, but they cannot be indexed or searched.";
```
Comment (javanna) [inline server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java:62]: Can you help me understand the user experience here? This seems like a deprecation warning, but it mentions that the field is not indexed nor searchable? How does that work then? We go ahead with the query and treat the field as unmapped?

Comment (benwtrent) [reply server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java:62]: They get an exception. Nothing can be indexed and searches will fail.

They will get an exception like: `"The [sparse_vector] field type is not supported on indices created on versions 8.0 to 8.10."`

Comment (javanna) [reply server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java:62]: do you mean before or after this change? if we fail why did we issue the deprecation?

Comment (benwtrent) [reply server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java:62]: I mean after this change. Before this change, they would receive a deprecation warning only if they created the index in 7.x and used the tech preview sparse_vector field.

They could NOT use sparse_vector in 8. 

But again, I can see how this change could effect indices that used sparse_vector in 7x and then they attempt to load that index via the archive index stuff :/

Comment (javanna) [reply server/src/main/java/org/elasticsearch/index/mapper/vectors/SparseVectorFieldMapper.java:62]: exactly :)

**Comment by javanna** [comment]
I went through this change another time, and I don't think we should make things as they are, to ensure compatibility with v7 indices. Does that make sense?

**Pull request closed without merging**